### ISSUE TITLE
DetailCurve.ByCurve does not work for Area Plans

### ISSUE THREAD

**Issue by frothey** [DynamoDS/DynamoRevit#1678]
If this issue is **not** a bug report or improvement request, please check the [Dynamo forum](https://forum.dynamobim.com/), and start a thread there to discuss your issue.

## 1.3.0.875

(Which version of Dynamo are you using? Go to Help > About if you're not sure.)

## 2018.0.1.2

(Which version of Revit are you using?)

## Operating system

Windows 10

## What did you do? 

Input Area Plan view and lines into DetailCurve.ByCurve. The same line input works with Floor Plan view. 

## What did you expect to see?

Detail lines drawn in the Area Plan View

## What did you see instead?

Warning: One or more of the input types are not matching. Couldn't find a version of ByCurve that takes arguments of type (Revit.Elements.Element,__array)
<img width="631" alt="2017-05-17_14-51-19" src="https://cloud.githubusercontent.com/assets/16231672/26178200/da2cff28-3b11-11e7-8831-004fc244f59e.png">

**Comment by eibre** [comment]
I  think this PR (https://github.com/DynamoDS/DynamoRevit/pull/1679) will fix the issue.

**Comment by ksobon** [comment]
@eibre yes, that particular error would be addressed by that PR, but it would generate a different error: 

![image](https://cloud.githubusercontent.com/assets/529781/26280237/d9764c86-3d98-11e7-8457-0c505944715d.png)

Detail Line creation is constrained to certain "annotation" view types. Area View is not one of them just yet. I will add that to the PR i submitted so that we can do this: 

![image](https://cloud.githubusercontent.com/assets/529781/26280265/f24f8d52-3d99-11e7-9667-4420b0bb1914.png)

@mjkkirschner is there any other view type that should be added to this list? 

```
internal bool IsAnnotationView()
        {
            switch (this.InternalView.ViewType)
            {
                case Autodesk.Revit.DB.ViewType.FloorPlan:
                case Autodesk.Revit.DB.ViewType.EngineeringPlan:
                case Autodesk.Revit.DB.ViewType.Detail:
                case Autodesk.Revit.DB.ViewType.Section:
                case Autodesk.Revit.DB.ViewType.Elevation:
                case Autodesk.Revit.DB.ViewType.CeilingPlan:
                case Autodesk.Revit.DB.ViewType.DraftingView:
                case Autodesk.Revit.DB.ViewType.AreaPlan:
                    return true;

                default: return false;
            }
        }
```

**Comment by frothey** [comment]
@ksobon Any timeframe on when area views will be added to DetailCurve.ByCurve?

**Comment by ksobon** [comment]
it's all in the same PR. so whenever that gets pulled in, it will be available in the next release. Or if you are adventurous you can pull down the source and build from branch that has the fix.

**Comment by smangarole** [comment]
@kronz and @mjkkirschner - Are we planning to merge this PR?

**Comment by johnpierson** [comment]
Closing this out - a fix for this was merged, as noted in the thread above. The issue was simply never closed.

If you still see this behaviour on a current build, please comment and we will reopen.
